medusa/providers/torrent/html/torrenting.py

```diff
@@ -0,0 +1,201 @@
+
+            # Skip column headers
+            for row in torrent_rows[1:]:
+                try:
+                    torrent_items = row.find_all('td')
+                    title = torrent_items[1].find('a').get_text(strip=True)
+                    download_url = urljoin(self.url, torrent_items[2].find('a')['href'])
+                    if not all([title, download_url]):
+                        continue
+
+                    seeders = try_int(torrent_items[5].get_text(strip=True))
+                    leechers = try_int(torrent_items[6].get_text(strip=True))
+
+                    # Filter unseeded torrent
+                    if seeders < min(self.minseed, 1):
+                        if mode != 'RSS':
+                            log.debug("Discarding torrent because it doesn't meet the"
+                                      " minimum seeders: {0}. Seeders: {1}",
+                                      title, seeders)
+                        continue
+
+                    torrent_size = torrent_items[4].get_text()
+                    size = convert_size(torrent_size) or -1
+
+                    pubdate_raw = torrent_items[1].find('div').get_text()
+                    pubdate = self.parse_pubdate(pubdate_raw, human_time=True)
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': pubdate,
+                    }
+                    if mode != 'RSS':
+                        log.debug('Found result: {0} with {1} seeders and {2} leechers',
+                                  title, seeders, leechers)
+
+                    items.append(item)
+                except (AttributeError, TypeError, KeyError, ValueError, IndexError):
+                    log.error('Failed parsing provider. Traceback: {0!r}',
+                              traceback.format_exc())
+
+        return items
+
+    def login(self):
+        """Login method used for logging in before doing search and torrent downloads."""
+        if dict_from_cookiejar(self.session.cookies).get('uid') and \
+                dict_from_cookiejar(self.session.cookies).get('pass'):
+            return True
+
+        if self.cookies:
+            self.add_cookies_from_ui()
+        else:
+            log.warning('Failed to login, you must add your cookies in the provider settings')
+            return False
+
+        login_params = {
+            'username': self.username,
+            'password': self.password,
+            'submit.x': 0,
+            'submit.y': 0,
+        }
+
+        response = self.session.post(self.urls['login'], data=login_params)
+        if not response or not response.text:
+            log.warning('Unable to connect to provider')
+            return False
+
+        if re.search('Username or password incorrect', response.text):
```

Comment:
Why would you need to use regex here? You could just as well check for the string in the text using `in`?